packages/hash/shared/src/entityStorePlugin.ts

```diff
@@ -64,6 +59,10 @@ export type EntityStorePluginAction = { received?: boolean } & (
       type: "updateEntityProperties";
       payload: {
         draftId: string;
+        metadata?: {
+          componentId: string;
+          dataEntity?: { [key: string]: unknown };
+        };
         properties: { [key: string]: unknown };
         merge: boolean;
       };
```

Comment:
Assuming it should only ever be one or the other.
```suggestion
      payload: {
        draftId: string;
        merge: boolean;
      } & (
        | {
            metadata?: {
              componentId: string;
              dataEntity?: { [key: string]: unknown };
            };
          }
        | { properties: { [key: string]: unknown } }
      );
    }
```

 I've commented elsewhere that we could make it clearer via naming where we're introducing special provision for entities referenced by blocks, which might result in something like:
```suggestion
      payload: {
        draftId: string;
        merge: boolean;
      } & (
        | {
            blockEntityMetadata?: {
              componentId: string;
              blockChildEntity?: { [key: string]: unknown };
            };
          }
        | { properties: { [key: string]: unknown } }
      );
    }
```

Reply:
It shouldn't always be one or the other, the metadata can be set at the same time as the properties

Reply:
A block entity only has `componentId` as a property, though (and a link to another entity). And no other entity can use `blockEntityMetadata`. What is the use case for setting both at once?

Reply:
My bad, i was looking at an instance where I had forgoten to remove the properties where I was using `blockEntityMetadata`. I'll update this soon!

Reply:
Addressed in 0087f8337a0b6cc0badb201a7b3e6c4a45710447
will rename field next